open_academy/models/session.py

```diff
@@ -1,14 +1,47 @@
-from odoo import models, fields
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
 
 
 class Session(models.Model):
     _name = "session"
     _description = "Session"
 
     name = fields.Char(required=True)
-    start_date = fields.Date()
+    start_date = fields.Date(default=lambda self: fields.Date.today(self))
     duration = fields.Float()
     seats = fields.Integer(string="Number of seats")
+    percent_seats = fields.Integer(string="Taken seats %", compute="_compute_seats")
+    active = fields.Boolean(default=True)
     instructor_id = fields.Many2one("res.partner", domain="[('instructor', '=', True)]")
     course_id = fields.Many2one("course")
     attendee_ids = fields.Many2many("res.partner")
+
+    @api.depends("seats", "attendee_ids")
+    def _compute_seats(self):
+        for record in self:
+            if record.seats > 0 and record.seats >= len(record.attendee_ids):
+                record.percent_seats = round(len(record.attendee_ids) / record.seats * 100)
+            else:
+                record.percent_seats = 0
+
+    @api.onchange("seats", "attendee_ids")
+    def _onchange_percent_seats(self):
+        if len(self.attendee_ids) > self.seats or self.seats < 0:
+            self.seats = self._origin.seats
+            self.attendee_ids = self._origin.attendee_ids
+
+            return {
+                "warning": {
+                    "title": _("Invalid Values"),
+                    "message": _("Cannot have a negative number of seats or more seats taken than the total number "
+                                 "of seats."),
+                }
+            }
+
+    @api.constrains("instructor_id", "attendee_ids")
+    def _check_instructor_in_attendees(self):
+        for record in self:
+            instructor_in_attendees = record.instructor_id in record.attendee_ids
+
+        if not instructor_in_attendees:
+            raise ValidationError(_("Instructor not present in the attendees of his/her own session"))
```

Comment:
Review this constraint, the exercise says:

`Add a constraint that checks that the instructor is not present in the attendees of his/her own session`, and I have the following error.

![image](https://user-images.githubusercontent.com/47753811/181607198-a3164c99-1907-4fe3-8883-f17e433e73ae.png)

You are confused, it is required to validate that the instructor is NOT in the set of elements of the assistants, in other words, an instructor can NOT be an assistant. On the other hand, it is not necessary to iterate over all the records, the first record that meets this criteria, should return the error